resources/views/pages/achievement/[achievement]/report-issue.blade.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+use function Laravel\Folio\{middleware, name};
+
+middleware(['auth', 'can:view,achievement']);
+name('achievement.report-issue');
+
+?>
+
+@props([
+    'achievement' => null, // Achievement
+])
+
+@php
+
+use App\Community\Enums\TicketType;
+
+$unlockedHardcore = false;
+$unlockedSoftcore = false;
+$unlockedManually = false;
+$hasHardcoreUnlocks = false;
+$hasSoftcoreUnlocks = false;
+$hasSession = false;
+
+$playerAchievements = request()->user()->playerAchievements()
+    ->join('Achievements', 'player_achievements.achievement_id', '=', 'Achievements.ID')
+    ->where('Achievements.GameID', $achievement->game->id)
+    ->get();
+
+$ticketType = TicketType::DidNotTrigger;
+
+foreach ($playerAchievements as $playerAchievement) {
+    if ($playerAchievement->achievement_id === $achievement->id) {
+        $unlockedHardcore = $playerAchievement->unlocked_hardcore_at !== null;
+        $unlockedSoftcore = $playerAchievement->unlocked_at !== null;
+        $unlockedManually = $playerAchievement->unlocker_id !== null;
+    }
+
+    $hasHardcoreUnlocks |= $playerAchievement->unlocked_hardcore_at !== null;
+    $hasSoftcoreUnlocks |= $playerAchievement->unlocked_at !== null;
+    $hasSession = true;
+}
+
+if ($unlockedHardcore || ($unlockedSoftcore && !$hasHardcoreUnlocks)) {
+    $ticketType = TicketType::TriggeredAtWrongTime;
+}
+
+if (!$hasSession) {
+    $hasSession = request()->user()->playerGames()->where('game_id', $achievement->game->id)->exists();
```

Comment:
Is this a good use case for `ActsAsPlayer::hasPlayed()`?

```php
request()->user()->hasPlayed($achievement->game);
```

Reply:
yep ;)